cmd/sync-server/main.go

```diff
@@ -0,0 +1,261 @@
+)
+
+// fetcher returns a channel of RepoProjects to be processed by workers. It uses the
+// "generator" concurrency pattern for feeding a channel.
+func fetcher(projectType, category string, bufferSize int, token chan struct{}, maxPages int) <-chan wporg.RepoProject {
+
+	// The new channel that will be processed by the workers.
+	out := make(chan wporg.RepoProject, bufferSize)
+
+	// Feed the channel concurrently.
+	go func(token chan struct{}) {
+
+		// Don't do anything until a token is available.
+		<-token
+
+		var response *wporg.ApiResponse
+		var err error
+
+		for {
+
+			page := 1
+			pages := 1
+
+			for page <= pages {
+
+				if projectType == "themes" {
+					response, err = client.RequestThemes(category, bufferSize, page)
+				} else {
+					response, err = client.RequestPlugins(category, bufferSize, page)
+				}
+
+				if err != nil {
+					log.Println(err)
+					page = pages + 1
+					continue
+				}
+
+				pages = response.Info.Pages
+				if maxPages != -1 && pages > maxPages {
+					pages = maxPages
+				}
+
+				// Themes.
+				for _, project := range response.Themes {
+					project.Type = projectType
+					out <- project
+				}
+
+				// Plugins.
+				for _, project := range response.Plugins {
+					project.Type = projectType
+					out <- project
+				}
+
+				page += 1
+			}
+
+			log.Println(projectType, "sync finished at", time.Now().Format(time.RFC3339))
+
+			// Finished, so pass on the token.
+			token <- struct{}{}
+
+			time.Sleep(time.Second * time.Duration(poolDelay))
+
+			// Wait until this process receives a token.
+			<-token
+		}
+	}(token)
+
+	return out
+}
+
+// pool starts a number of infoWorkers. This uses a "worker" pattern where each worker will
+// read from a projects channel to process the results.
+func pool(workers int, projects <-chan wporg.RepoProject, checker sync.UpdateChecker, dispatcher sync.Dispatcher, messages chan *message.Message) {
+	for i := 0; i < workers; i++ {
+		go infoWorker(projects, checker, dispatcher, messages)
+	}
+}
+
+// infoWorker reads a project from the projects channel, runs it through the update check
+// and (conditionally) sends it to the dispatcher which loads up the job queue.
+func infoWorker(projects <-chan wporg.RepoProject, checker sync.UpdateChecker, dispatcher sync.Dispatcher, messages chan *message.Message) {
+	for {
+		select {
+		case project := <-projects:
+			// If this project is not in sync (or forced), then sent it to the queue.
+			if checker.UpdateCheck(project) || forcedSync {
+				if err := dispatcher.Dispatch(project); err == nil {
+					// If its been successfully added to the queue then change the
+					// project state.
+					checker.RecordUpdate(project)
+				} else {
+					log.Println(project.Name, err)
+				}
+			}
+		}
+	}
+}
+
+// defaultStandards gets the standards to run against a project.
+func defaultStandards() []string {
+	return []string{
+		"phpcs_wordpress",
+		"phpcs_phpcompatibility",
```

Comment:
What about lighthouse?

Reply:
we can add that, but it should also be added below to `defaultAudits`...  once we sort out the audit server I'd like to review this.

Reply:
Does Lighthouse have any real options being passed to it?

Reply:
We'll double back on this once we've identified what the options would be.